### PR TITLE
Refactor naming of config options and descriptions.

### PR DESCRIPTION
**Pull request by simitt**

### CODE REVIEW

File: _meta/beat.reference.yml [hunk 1]
```diff
@@ -4,51 +4,62 @@
...
   #secret_token:
-
   #ssl.enabled: false
   #ssl.certificate : "path/to/cert"
   #ssl.key : "path/to/private_key"
 
+  # Please be aware that frontend support is an experimental feature at the moment!
   #frontend:
-    # Please be aware that frontend support is an experimental feature at the moment!
+    # Set enabled to true for becoming an early tester of frontend support.
```
Comment (roncohen) [inline _meta/beat.reference.yml]: `To enable experiment frontend support, set this to true. `

File: _meta/beat.reference.yml [hunk 2]
```diff
@@ -4,51 +4,62 @@
...
     #enabled: false
 
-    # Rate Limit per second and IP address
+    # Rate Limit per second and IP address for requests sent to the frontend endpoint.
```
Comment (roncohen) [inline _meta/beat.reference.yml]: lower case `L` in `limit`

File: _meta/beat.reference.yml [hunk 3]
```diff
@@ -4,51 +4,62 @@
...
+    # If an item in the list is a single '*', everything will be allowed
+    #allow_origins : ['*']
 
-    # Regexp to be matched against a stacktrace frame's `file_name` and `abs_path` attributes,
+    # Regexp to be matched against a stacktrace frame's `file_name` and `abs_path` attributes.
     # If the regexp matches, the stacktrace frame is considered to be a library frame.
     #library_pattern: "node_modules|bower_components|~"
 
-    #sourcemapping:
+    # If a mapping Source Map is found in Elasticsearch, Source Mapping is automatically applied
```
Comment (roncohen) [inline _meta/beat.reference.yml]: source map should not be capitalized, `source map` is fine.

Comment (roncohen) [reply _meta/beat.reference.yml]: `If a source map has previously been uploaded, source mapping is automatically applied ...`

File: _meta/beat.reference.yml [hunk 4]
```diff
@@ -4,51 +4,62 @@
...
+    #source_mapping:
 
       # Source Maps are are fetched from Elasticsearch and then kept in an in-memory cache for a certain time.
       # The `cache.expiration` determines how long a Source Map should be cached before fetching it again from Elasticsearch.
       # Note that values configured without a time unit will be interpreted as seconds.
       #cache:
         #expiration: 5m
 
-      # Sourcemaps are stored in the same index as transaction and error documents.
+      # Source Maps are stored in the same index as transaction and error documents.
```
Comment (roncohen) [inline _meta/beat.reference.yml]: lower case `source map`

**Comment by simitt** [comment]
@roncohen thanks for the feedback, please see https://github.com/elastic/apm-server/pull/504.

**Pull request merged**